### ISSUE TITLE
Support for realm keys data resource

### ISSUE THREAD

**Issue by blakebarnett** [crossplane-contrib/provider-keycloak#119]
The terraform provider has support for outputting the realm keys: https://registry.terraform.io/providers/mrparkers/keycloak/latest/docs/data-sources/realm_keys

It would be very useful to have this available with support for `.spec.writeConnectionSecretToRef` so we can automate based on the content.

**Comment by Breee** [comment]
There are no real datasources in crossplane. 
You can import resources managed-only tho. 

So I'll have to check out how they can be imported

**Comment by art10m** [comment]
Are there any updates or known solutions for importing realm keys into Crossplane-managed Keycloak resources?

**Comment by Breee** [comment]
Nope, there is still no way to import datasources in crossplane sadly. 
We could: 
-  Design our own CRD / Crossplane for Realm keys to just pull them into a secret tho. I have no experience with writing custom controllers in crossplane yet tho.
- Or:  I think a function similar to  https://gitlab.com/corewire/images/crossplane/function-keycloak-builtin-objects/-/tree/main?ref_type=heads  could be defined fairly quick.  I mean in general that should be pretty easy to do, just fetching the realmkeys and saving into a secret 
- As a workaround you could use   https://marketplace.upbound.io/providers/upbound/provider-terraform/v0.19.4  and import the realm-keys that way.

**Comment by Breee** [comment]
Nowadays crossplane still does not Support that, Best solution is to use the opentofu Provider I guess and write it into a secret
